libraries/stdlib/samples/test/samples/collections/iterators.kt

```diff
@@ -0,0 +1,71 @@
+/*
+ * Copyright 2010-2017 JetBrains s.r.o.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package samples.collections
+
+import samples.Sample
+import samples.assertPrints
+import java.util.*
+
+class Iterators {
+
+    @Sample
+    fun iteratorForEnumeration() {
+        val list = arrayListOf<String>()
+        val vector = Vector<String>().apply {
+            add("RED")
+            add("GREEN")
+            add("BLUE")
+        }
+        val iterator = vector.elements().iterator()
+        while (iterator.hasNext()) {
+            list.add(iterator.next())
+        }
+
+        assertPrints(list.joinToString(":"), "RED:GREEN:BLUE")
+    }
+
+    @Sample
+    fun iterator() {
+        val intIterator = IntProgression.fromClosedRange(1, 3, 1).iterator()
+        val iterator = intIterator.iterator()
```

Comment:
There's no point in calling `iterator()` when you already have an iterator. Usually this extension is used to satisfy `for` operator convention for an Iterator. For example you can get a MutableIterator from a MutableCollection, then iterate it with `for` operator and do something with it at the same time, namely remove some items.